dts/common/yaml/zephyr-i2c.yaml

```diff
@@ -0,0 +1,19 @@
+---
+title: Zephyr specific I2C properties
+id: i2c
+version: 0.1
+
+description: >
+    This binding gives Zephyr specific I2C properties
+
+properties:
+  - zephyr,speed :
+      type: int
+      category: required
+      description: Speed setting (1..5) as defined in i2c.h
+      generation: define
+  - zephyr,use-10-bit-addr :
```

Comment:
I believe the current support for 10-bit addressing in i2c.h file is incorrect. I2C_ADDR_10_BITS flag is supposed to be used with i2c_configure command and set only once. We treat it the same as speed, master/slave mode. i.e. it's a property which applies to all transactions performed on the bus. That's wrong. From I2C specification:
"Devices with 7-bit and 10-bit addresses can be connected to the same I2C-bus, and both 7-bit and 10-bit addressing can be used in all bus speed modes."
It means that I2C_ADDR_10_BITS should be a property of i2c_msg.flags. In that case every message sent on the I2C bus can have independently set addressing mode.

As such the zephyr,use-10-bit-addr property should be removed from DTS.